cdap-ui/app/cdap/components/KeyValuePairs/index.js

```diff
@@ -98,6 +109,12 @@ export default class KeyValuePairs extends Component {
   }
 
   shouldComponentUpdate(nextProps) {
+    if (this.state.pairs.length !== nextProps.keyValues.pairs.length) {
```

Comment:
Will this update if there if user just editing existing keyvalue? since the length is the same

Reply:
That will be taken care of in the `KeyValueStore` and `KeyValuePair` thats how we retain the focus on textboxes when typing in it because the input boxes are redux-connected